cmd/cosign/cli/verify/verify_blob.go

```diff
@@ -178,10 +186,20 @@ func VerifyBlobCmd(ctx context.Context, ko options.KeyOpts, certRef, certEmail,
 			// check if cert is actually a public key
 			co.SigVerifier, err = sigs.LoadPublicKeyRaw(certBytes, crypto.SHA256)
 		} else {
+			if certChain == "" {
+				co.RootCerts, err = fulcio.GetRoots()
+				if err != nil {
+					return fmt.Errorf("getting Fulcio roots: %w", err)
+				}
+				co.IntermediateCerts, err = fulcio.GetIntermediates()
+				if err != nil {
+					return fmt.Errorf("getting Fulcio intermediates: %w", err)
+				}
+			}
 			co.SigVerifier, err = cosign.ValidateAndUnpackCert(cert, co)
```

Comment:
Should we handle `else` for when the cert chain is set? Otherwise, the certificate wouldn't validate if it's been signed by a non-fulcio pki

Reply:
good catch!

Reply:
lemme see if I can also add a test here.

Reply:
Hm...I think it's pretty unexpected that `--certificate-chain` would mean "accept whatever cert was at the root of the chain."

This is part of why I think an explicit `--certificate-root` flag would be a good idea.

Reply:
We do document the the chain is the intermediates and root - https://github.com/sigstore/cosign/blob/04f509c7c95538d50a6065c2f098eada59d9f139/cmd/cosign/cli/options/certificate.go#L67-L70

Reply:
done! thank you. test correctly regressed before, succeeds with change.

Reply:
TBH that help text doesn't imply to me that "we'll interpret whatever is in the last spot as the root certificate," it's ambiguous whether the tool will compare it to some other source of truth.

Plus, with `COSIGN_EXPERIMENTAL=1` set, we *do* do that check, right? Isn't that what https://github.com/sigstore/cosign/issues/1554 is about?

Reply:
I thought we consider self-signed certs as roots?

Reply:
Ugh we don't, but that's what the SIGSTORE_ROOT_FILE code does.

Can we file a separate issue on that?

Reply:
If we're sure that this matches the behavior from `v1.11.x`, and we explain it more clearly in the help text, I'm okay with dealing with this problem later.

Reply:
Yea, let's. It's a bit unclear to me what should be the right behavior here, especially if I provide both a sigstore root and certificate chain.

Reply:
The behavior here should match 1.11.x (the more significant change is around checking expiration)

Reply:
If someone could help me update the release notes to close this out, that'd be great. I added one in. I'm not sure if anything breaks.

Reply:
Hm I don't see anything under "Release Notes" in the PR description.

Maybe:

> The `v1.12.0` release introduced a regression: when `COSIGN_EXPERIMENTAL` was not set, `cosign verify-blob` would check a `--certificate` (without a `--certificate-chain` provided) against the operating system root CA bundle. In this release, Cosign checks the certificate against Fulcio's CA root instead (restoring the earlier behavior).


I think we should have a summary of the current behavior (as opposed to a diff, in the release note) somewhere too; maybe that's what we can hash out in https://github.com/sigstore/cosign/pull/2254

Reply:
Sorry, I put it in the summary instead. Moved, but feel free to edit.

Sg, I'll add a longer explanation